src/main/java/dinkplugin/notifiers/DiaryNotifier.java

```diff
@@ -62,8 +64,13 @@ public void onVarbitChanged(VarbitChanged event) {
         if (id < 0) return;
         Pair<String, AchievementDiary.Difficulty> diary = DIARIES.get(id);
         if (diary == null) return;
-        if (diaryCompletionById.isEmpty()) return;
         if (!super.isEnabled()) return;
+        if (diaryCompletionById.isEmpty()) {
+            if (client.getGameState() == GameState.LOGGED_IN && isComplete(id, event.getValue())) {
+                log.info("Skipping {} {} diary completion that occurred before map initialization", diary.getRight(), diary.getLeft());
+            }
+            return;
+        }
 
         int value = event.getValue();
         Integer previous = diaryCompletionById.get(id);
```

Comment:
I'd like a log entry + comment in this case.

What does it mean when `previous == null` is true?
What does it mean when `value < previous`?

Reply:
Added in https://github.com/pajlads/DinkPlugin/pull/129/commits/20a5c22b9a9a65f847643f4b40bf0a571615a22c - these cases shouldn't occur because we reset the diary map on game state change, but it's a safeguard for something going wrong on runelite's side